examples/servers/simple-auth/mcp_simple_auth/auth_server.py

```diff
@@ -0,0 +1,259 @@
+        Proxy endpoint to get GitHub user info using stored GitHub tokens.
+
+        Resource Servers call this with MCP tokens to get GitHub user data
+        without exposing GitHub tokens to clients.
+        """
+        try:
+            # Extract Bearer token
+            auth_header = request.headers.get("authorization", "")
+            if not auth_header.startswith("Bearer "):
+                return JSONResponse({"error": "unauthorized"}, status_code=401)
+
+            mcp_token = auth_header[7:]
+
+            # Get GitHub user info using the provider method
+            try:
+                user_info = await oauth_provider.get_github_user_info(mcp_token)
+                return JSONResponse(user_info)
+            except ValueError as e:
+                if "No GitHub token found" in str(e):
+                    return JSONResponse({"error": "no_github_token"}, status_code=404)
+                elif "GitHub API error" in str(e):
+                    return JSONResponse({"error": "github_api_error"}, status_code=502)
+                raise
+
+        except Exception as e:
+            logger.exception("GitHub user info error")
+            return JSONResponse({"error": str(e)}, status_code=500)
+
+    routes.append(
+        Route(
+            "/github/user",
+            endpoint=cors_middleware(github_user_handler, ["GET", "OPTIONS"]),
+            methods=["GET", "OPTIONS"],
+        )
+    )
+
+    return Starlette(debug=True, routes=routes)
+
+
+async def run_server(settings: AuthServerSettings):
+    """Run the Authorization Server."""
+    auth_server = create_authorization_server(settings)
+
+    config = Config(
+        auth_server,
+        host=settings.host,
+        port=settings.port,
+        log_level="info",
+    )
+    server = Server(config)
+
+    logger.info("=" * 80)
+    logger.info("MCP AUTHORIZATION SERVER")
+    logger.info("=" * 80)
+    logger.info(f"Server URL: {settings.server_url}")
+    logger.info("Endpoints:")
+    logger.info(f"  - OAuth Metadata: {settings.server_url}/.well-known/oauth-authorization-server")
+    logger.info(f"  - Client Registration: {settings.server_url}/register")
+    logger.info(f"  - Authorization: {settings.server_url}/authorize")
+    logger.info(f"  - Token Exchange: {settings.server_url}/token")
+    logger.info(f"  - Token Introspection: {settings.server_url}/introspect")
+    logger.info(f"  - GitHub Callback: {settings.server_url}/github/callback")
+    logger.info(f"  - GitHub User Proxy: {settings.server_url}/github/user")
+    logger.info("")
+    logger.info("Resource Servers should use /introspect to validate tokens")
+    logger.info("Configure GitHub App callback URL: " + settings.github_callback_path)
+    logger.info("=" * 80)
+
+    await server.serve()
+
+
+@click.command()
+@click.option("--port", default=9000, help="Port to listen on")
+@click.option("--host", default="localhost", help="Host to bind to")
+def main(port: int, host: str) -> int:
```

Comment:
This will give a lot of maintenance burden. People will ask a lot of uvicorn's parameters to be added here.

Reply:
Ah, my bad. All good here.